internal/comparisons/comparisons.go

```diff
@@ -0,0 +1,102 @@
+package comparisons
+
+import (
+	"cmp"
+	"slices"
+	"strings"
+)
+
+// IntValue returns the value of a given int pointer
+// or 0 if the pointer is nil
+func intValue(i *int) int {
+	if i != nil {
+		return *i
+	}
+	return 0
+}
+
+// StringValue returns the value of a given string pointer
+// or empty string if the pointer is nil
+func stringValue(s *string) string {
+	if s != nil {
+		return *s
+	}
+	return ""
+}
+
+// isEmpty returns true if the given value is empty
+func IsEmpty(v interface{}) bool {
+	switch v := v.(type) {
+	case int:
+		return v == 0
+	case *int:
+		return intValue(v) == 0
+	case string:
+		return v == ""
+	case *string:
+		return stringValue(v) == ""
+	case nil:
+		return true
+	default:
+		return false
+	}
+}
+
+// Subsets returns true if the first slice is a subset of the second slice
```

Comment:
Compared to the original, this function uses generics:

https://github.com/equinix/terraform-provider-equinix/blob/3cfe64eca7a8964427a6eda9a97c71eaf60ca8ab/equinix/provider.go#L197-L204